pkg/distributor/distributor.go

```diff
@@ -92,6 +94,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	flag.Float64Var(&cfg.IngestionRateLimit, "distributor.ingestion-rate-limit", 25000, "Per-user ingestion rate limit in samples per second.")
 	flag.IntVar(&cfg.IngestionBurstSize, "distributor.ingestion-burst-size", 50000, "Per-user allowed ingestion burst size (in number of samples).")
 	flag.BoolVar(&cfg.HealthCheckIngesters, "distributor.health-check-ingesters", false, "Run a health check on each ingester client during periodic cleanup.")
+	flag.Var(&cfg.DontShardMetricNamesFrom, "distributor.dont-shard-metric-names-from", "The date & time after which we stop sharding by metric name.")
```

Comment:
Why does it need to be a time and not just a timeless switch? We're querying everything anyways?

Reply:
The problem I was trying to solve was that roll-outs the distributors isn't instantaneous - during the process queries would return inconsistent results.   By making it a switch, this should be minimised to clock skew.  Might not be worth it though.  WDYT?